src/map_command_handler.hpp

```diff
@@ -190,9 +190,30 @@ class map_command_handler
 		}
 		else if (help_on_unknown_) {
 			utils::string_map symbols;
+		    // Get the input command on a string
+            std::string string_user_ = get_cmd();
+            // Get all commands (even if disallowed)
+            std::vector<std::string> command_list_ = get_commands_list();
+            // Initialize the distance and the command_proposal symbol so it
+            // does not appear if no candidate is found.
+            symbols["command_proposal"] = "";
+            int distance_ = 0;
+            // Compare the input with every command (excluding alias)
+            for (typename command_map::value_type i : command_map_) {
```

Comment:
Didn't Wesnoth's style guide say no spaces before the parenthesis in control statements? I know the actual code is inconsistent though…

Reply:
I copied the convention from `for (typename command_map::value_type i : command_map_) {` (same file, line 224) So I'm unsure. I'll search for the actual convention guide.

Reply:
Yeah, I think what happened is we decided to change the style guide at some point but no-one updated existing code to follow it… either that, or we didn't _have_ a style guide and finally made one and no-one updated existing code.

Reply:
Hmmm I wouldn't mind fitting these files to convention. Not in this PR tho. I'm gonna guess there isn't a tool like wmlindent but for C++?

Reply:
There is ./utils/fix_whitespace.sh, which runs wmlindent and also adjusts whitespace in C++ and Lua files, but that only covers leading whitespace (indentation), and the non-wmlindent part I believe just converts tabs to spaces or spaces to tabs rather than actually fixing misindented code.

Besides that, I think there's a clang-format specification in the repository that you could use, though I'm not sure when it was last updated (@Vultraz?).

Reply:
`clang-format` is up-to-date.

Reply:
`clang-format` is up-to-date, but is also a tool with a user-unfriendly interface. For project-specific settings it doesn't take the options file on the command line, instead it has to read it in the current working directory.

1. Make sure everything's checked in to Git, in case a tonne of unrelated changes appear.
2. In a command prompt, change to Wesnoth's `src` directory (not the root, the directory with the `.clang-format` file in it)
3. Run `clang-format --style=file -i file_to_format.cpp`
    - `--style=file` is entered verbatim, don't change that for a filename
    - The `-i` makes it overwrite the file (and also means you can give multiple files in one run), otherwise it outputs to stdout

Reply:
If you use VSCode the Format Document/Format Selection command uses `clang-format`. No need to use the command prompt.

Reply:
Thanks, I'll have a look at that. I had to have a look at VSCode anyways because of the Msys sdl issue so I may finally change my building tool.